CodePush.m

```diff
@@ -362,11 +366,19 @@ - (void)savePendingUpdate:(NSString *)packageHash
 /*
  * This is native-side of the RemotePackage.download method
  */
-RCT_EXPORT_METHOD(downloadUpdate:(NSDictionary*)updatePackage
+RCT_EXPORT_METHOD(downloadUpdate:(NSDictionary*)immutableUpdatePackage
                         resolver:(RCTPromiseResolveBlock)resolve
                         rejecter:(RCTPromiseRejectBlock)reject)
 {
     dispatch_async(dispatch_get_main_queue(), ^{
+        NSDictionary* updatePackage = [immutableUpdatePackage mutableCopy];
+        NSUserDefaults *preferences = [NSUserDefaults standardUserDefaults];
```

Comment:
The way the 
```
+ (NSURL *)bundleURLForResource:(NSString *)resourceName
                  withExtension:(NSString *)resourceExtension
```
is designed, it looks like the resource name and resource extension is variable and may change. Therefore the only place that knows how to find the binary timestamp is in that method, that's why I thought of saving the timestamp in the defaults inside that method so that other methods can retrieve it.